UofS-Pulse-Binfo/tripal_genetic
# QTL Importer

This importer adds QTL data to an existing map.

## Testing
1. Clone this repository, install it and Tripal Map and spin up phpunit/tripaltestsuite: `composer up`
2. Load a fake map for testing purposes: `./vendor/bin/tripaltest db:seed GeneticMapSeeder`
3. Run automated testing and ensure it passes: `./vendor/bin/phpunit`
4. Create cvterms to match any traits you have in your test file. For `tests/example_files/qtl.singletrait.tsv` you would create a term with the name `Days to Flowering`. You can create cvterms through the UI at Admin > Tripal > Data Loaders > Controlled Vocabularies > Manage Chado CV Terms. Note the CV you create it in as you will have to select it in the importer.
5. In your site, go to Admin > Tripal > Data Loaders > QTL Importer. Choose your fake map created in step 2 and the CV you noted down in step 4.
6. Upload a test file of your choice or use `tests/example_files/qtl.singletrait.tsv`
7. Choose submit and run the Tripal Job
8. Go to Admin > Tripal > Data Storage > Chado > Materialized Views and sync both `tripal_map_genetic_markers_mview` and `tripal_map_qtl_and_mtl_mview`
9. View your QTL using Tripal Map at `localhost/MapViewer`
